Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs

```diff
@@ -557,12 +569,22 @@ public bool RegisterService(Type type, IMixedRealityService service)
         /// </summary>
         /// <typeparam name="T">The interface type for the system to be registered.</typeparam>
         /// <param name="concreteType">The concrete type to instantiate.</param>
+        /// <param name="supportedPlatforms">The runtime platform to check against when registering.</param>
         /// <param name="args">Optional arguments used when instantiating the concrete type.</param>
         /// <returns>True, if the service was successfully registered.</returns>
-        public bool RegisterService<T>(Type concreteType, params object[] args)
+        public bool RegisterService<T>(Type concreteType, SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1), params object[] args)
```

Comment:
How were services being limited to platforms previously?  Does this duplicate work?

Reply:
Mostly asking because I do not see (in this PR) a removal of the old filtering methology

Reply:
All systems should be platform agnostic.

Extension services may or may not be platform specific.

Data providers are platform specific.

Previously we were iterating over the list of extension services and filtering in the for loop.

Reply:
@davidkline-ms removal of the old filtering methodology [is here](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3173/files/633623d5b39bacd549ebc8f72bdf94a93e41b483#diff-f21536a45f2f15aa366accfb4e1e7d9bL222)

Reply:
Thanks, That's the section I was referring to

Reply:
Thanks for pointing that out @keveleigh. I knew I was scanning over something without noticing it. :)